src/storage/txn/scheduler.rs

```diff
@@ -1069,10 +1075,18 @@ impl<E: Engine, L: LockManager> TxnScheduler<E, L> {
             .unwrap();
     }
 
-    fn is_undetermined_error(_e: &tikv_kv::Error) -> bool {
-        // TODO: If there's some cases that `engine.async_write` returns error but it's
-        // still possible that the data is successfully written, return true.
-        false
+    // Return true if raftstore returns error and the underlying write status could
+    // not be decided.
+    fn is_undetermined_error(e: &tikv_kv::Error) -> bool {
```

Comment:
This function is reserved for controlling fair locking's behavior. As the undetermined error is just introduced, I think we may need to consider add more related tests for fair locking later.

Reply:
Good idea.